delete_purchases/delete_purchases.py

```diff
@@ -39,51 +28,83 @@ def get_purchase_lines(api):
     request_body = {"query": queries.GET_PURCHASE_LINES}
     return api.request(request_body)["data"]
 
+def get_purchase_line_etag(id,api):
+    request_body = {"query": queries.GET_PURCHASE_LINE_ETAG, "variables": {"id": id}}
+    purchase_line = api.request(request_body)["data"]
+    _etag = purchase_line["purchaseOrderLine"]["_etag"]
+    return _etag
 
 def get_purchases(api):
     request_body = {"query": queries.GET_PURCHASES}
     return api.request(request_body)["data"]
 
+def build_list_aboms_items(purchase_order_lines):
+    list_of_pos = set()
+    for purchase_order_line in purchase_order_lines["purchaseOrderLines"]["edges"]:
+        if "partInventories" in purchase_order_line["node"]:
+            if any(part_inventory["installed"] for part_inventory in purchase_order_line["node"]["partInventories"]) or any(part_inventory["kitted"] for part_inventory in purchase_order_line["node"]["partInventories"])or any(part_inventory["received"] for part_inventory in purchase_order_line["node"]["partInventories"]):
+                po_id = purchase_order_line["node"]["purchaseOrder"]["id"]
+                list_of_pos.add(po_id)
+    return list(list_of_pos)
+
+def build_list_receipts(receipts):
+    LIST_OF_POS = []
```

Comment:
Should this be called list_of_receipts?

Reply:
@weirandrewr

Reply:
no, because I'm finding POs that have receipts on them.